net: lib: Pull out and generalize Azure MQTT library (#9884)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

Libraries for networking
------------------------

> * Added :ref:`lib_mqtt_helper` library. Convenience library that simplifies Zephyr MQTT API and socket handling.

Reviewer: ```suggestion
* Added :ref:`lib_mqtt_helper` library, that simplifies Zephyr MQTT API and socket handling.
```

* :ref:`lib_azure_iot_hub` library:

>   * Pulled out :file:`azure_iot_hub_mqtt.c` and created a new library :ref:`lib_mqtt_helper` that the library now implements.

Reviewer: ```suggestion
  * Pulled out :file:`azure_iot_hub_mqtt.c`  that is now implemented by a new library :ref:`lib_mqtt_helper` .
```

>   * Pulled out :file:`azure_iot_hub_mqtt.c` and created a new library :ref:`lib_mqtt_helper` that the library now implements.

Reviewer: ```suggestion
  * Pulled out :file:`azure_iot_hub_mqtt.c`, which is now being implemented by a new library :ref:`lib_mqtt_helper` .
```
